scalanet/src/io/iohk/scalanet/peergroup/dynamictls/CustomTlsValidator.scala

```diff
@@ -109,16 +153,48 @@ object CustomTlsValidator {
     * The certificate MUST contain the libp2p Public Key Extension. If this extension is missing,
     * endpoints MUST abort the connection attempt
     *
+    * Endpoints MUST abort the connection attempt if the certificate contains critical extensions
+    * that the endpoint does not understand.
     */
   private def validateCertificateExtension(
       cert: X509Certificate,
       extensionId: String
   ): Either[CertificateError, Array[Byte]] = {
-    val extension = cert.getExtensionValue(extensionId)
-    if (extension == null) {
-      Left(NoCertExtension)
-    } else {
-      Right(extension)
+    Option(cert.getExtensionValue(extensionId)).toRight(NoCertExtension).flatMap { extension =>
+      val criticalExtensions = cert.getCriticalExtensionOIDs
```

Comment:
So original spec does not say if this extension should be critical or not. In our project we always mark this extension as critical, but in theory someone could take `scalanet` library and modify it to connect to us without it being marked as criticial. So in theory it can happen, that we will get this extension as non critical one.